### ISSUE TITLE
Models: documented public methods/attributes to extract best parameters for Auto* models

### ISSUE THREAD

**Issue by yarnabrina** [Nixtla/statsforecast#829]
### Description

Especially for the auto models it is quite common for users to know which parameter combination is the best model for reporting/analyses purposes, and also to know what are the corresponding AIC/BIC/etc. values. So, something along the lines of the following will be useful:

1. `get_best_parameters()` or `best_paramaters`
2. `get_best_model_score` or `best_model_score`

### Use case

This is a request raised multiple times by `sktime` users for the `_GeneralisedStatsForecastAdapter` [adapter](https://github.com/sktime/sktime/blob/main/sktime/forecasting/base/adapters/_generalised_statsforecast.py). Using this adapter, we have exposed all the 5 auto models: `AutoARIMA`, `AutoTheta`, `AutoETS`, `AutoCES`, `AutoTBATS`, and a few non-auto models.

As of now, `model_` attribute gives few details, but the keys seem to vary for different algorithms and it is difficult to expose parameters/scores reliably for all models using this.

**Comment by yarnabrina** [comment]
@AzulGarza @jmoralez , can you please comment on this issue? Is it safe to add a `get_fitted_params` methon in `statsforecast` wrapper of `sktime` relying on `model_` attribute, or will you recommend something else?

The same question also applies for `neuralforecast` probably.

**Comment by yarnabrina** [comment]
> Just to be sure, these would return the values that are tuned, right? For example for `AutoETS` it would return `model` and `damped`.

Yes, right. That'll help users to specify those themselves in future in case they want to avoid the tuning.

**Comment by Arnold-Kakas** [comment]
Hi all,

Any update on this point? This would be great to have available.
